File: 20160712/20160712_1.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
	# when len == 0 (s = '' or s = ' '), IndexError occurs
	ws = s.split()
	return len(ws[-1]) if len(ws) > 0 else 0

s = Solution()
print "%20s %8s %8s" % ("Input", "Output", "Expected")
print "%20s %8d %8d" % ('""', s.lengthOfLastWord(""), 0)
print "%20s %8d %8d" % ('" "', s.lengthOfLastWord(" "), 0)
print "%20s %8d %8d" % ('"a "', s.lengthOfLastWord("a "), 1)
print "%20s %8d %8d" % ("Hello World", s.lengthOfLastWord("Hello World"), 5)

File: 20160712/20160712_1.c
#include<string.h>
#include<stdio.h>

int lengthOfLastWord(char* s) {
    int i = strlen(s) - 1;
    int c = 0;
    while(s[i--] == ' ');
    i++;
    for(;i>=0;i--){
        if(s[i] == ' ')
            break;
        else
            c++;
    }
    return c;
}

int main()
{
    printf("%20s %8s %8s\n", "Input", "Output", "Expected");
    printf("%20s %8d %8d\n", "\"\"", lengthOfLastWord(""), 0);
    printf("%20s %8d %8d\n", "\" \"", lengthOfLastWord(" "), 0);
    // the output is 0, the solution is 1. That's why we need trim first.
    printf("%20s %8d %8d\n", "\"a \"", lengthOfLastWord("a "), 1);
    printf("%20s %8d %8d\n", "Hello World", 
        lengthOfLastWord("Hello World"), 5);
    return 0;
}
